services/galley/src/Galley/External/LegalHoldService.hs

```diff
@@ -167,3 +244,43 @@ makeLegalHoldServiceRequest tid reqBuilder = do
     mkReqBuilder token =
       reqBuilder
         . Bilge.header "Authorization" ("Bearer " <> toByteString' token)
+
+versionToInt :: LhApiVersion -> Int
+versionToInt V0 = 0
+versionToInt V1 = 1
+
+versionToBS :: LhApiVersion -> ByteString
+versionToBS = ("v" <>) . BS8.pack . show . versionToInt
+
+versionedPaths :: LhApiVersion -> [ByteString] -> Http.Request -> Http.Request
+versionedPaths V0 paths = Bilge.paths paths
+versionedPaths v paths = Bilge.paths (versionToBS v : paths)
+
+supportedClientVersions :: Set LhApiVersion
+supportedClientVersions = Set.fromList [minBound .. maxBound]
+
+-- | Find the highest common version between wire-server and the legalhold service.
+-- If the legalhold service does not support the `/api-version` endpoint, we assume it's `v0`.
+-- If there is no common version, we log a warning and use `v0`. This violates the fail-fast principle,
+-- but makes multiple API endpoints backwards compatible.
```

Comment:
which endpoints? how?  doesn't it work without this?  `getLegalHoldApiVersion` returns `Nothing` (because end-point doesn't exist), so we end up in line 277 below, no?